src/duct-waf-workspace/duct-waf.repository.ts

```diff
@@ -7,13 +7,17 @@ export class DuctWafWorkspaceRepository extends Repository<DuctWaf> {
   async getDuctWafByLocIdBDateBHourWafValue(
     locationId: string,
     wafBeginDate: Date,
+    wafEndDate: Date,
     wafBeginHour: number,
+    wafEndHour: number,
     wafValue: number,
   ): Promise<DuctWaf> {
     return this.createQueryBuilder('dw')
       .where('dw.locationId = :locationId', { locationId })
       .andWhere('dw.wafBeginDate = :wafBeginDate', { wafBeginDate })
       .andWhere('dw.wafBeginHour = :wafBeginHour', { wafBeginHour })
+      .andWhere('dw.wafEndDate = :wafEndDate', { wafEndDate })
```

Comment:
The End Date and Hour should only be added to the where clause if EndDate (the method param) is not null.  And the resulting Where clause should look like:

If End Date is Null:
    ... AND (wafBeginDate = :wafBeginDate AND wafBeginHour = :wafBeginHour)

If End Date is not Null:
   ... AND ( (wafBeginDate = :wafBeginDate AND wafBeginHour = :wafBeginHour) OR (wafEndDate = :wafEndDate AND wafEndHour = :wafEndHour) )